src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -117,6 +118,25 @@ public virtual string CreateToken(string payload, SigningCredentials signingCred
             return CreateTokenPrivate(JObject.Parse(payload), signingCredentials, null, null);
         }
 
+        /// <summary>
+        /// Creates a JWS(Json Web Signature).
+        /// </summary>
+        /// <param name="tokenDescriptor">A <see cref="SecurityTokenDescriptor"/> that contains details of contents of the token.</param>
+        /// <returns>A JWS in Compact Serialization Format.</returns>
+        public virtual string CreateToken(SecurityTokenDescriptor tokenDescriptor)
+        {
+            if (tokenDescriptor == null)
+                throw LogHelper.LogArgumentNullException(nameof(tokenDescriptor));
+
+            if (tokenDescriptor.Payload == null || tokenDescriptor.Payload.Count == 0)
```

Comment:
I would recommend using the Any() linq expression (e.g. || !tokenDescriptor.Payload.Any()) because it will enumerate at most one item in the collection. The Count property is cheap if the collection has already been evaluated, but will otherwise require all items in the collection to be enumerated which could be expensive if the collection is large.

Reply:
@mafurman 	@CBaud 	+1. Yes count requires the underlying structure to sometimes do a lot of work.

Reply:
I didn't realize this. Good suggestion! 👍